apps/astarte_appengine_api/lib/astarte_appengine_api/auth/auth.ex

```diff
@@ -19,11 +19,15 @@
 defmodule Astarte.AppEngine.API.Auth do
   alias Astarte.AppEngine.API.Queries
   alias Astarte.DataAccess.Database
+  alias Astarte.AppEngine.API.Config
+  alias Astarte.Core.CQLUtils
 
   require Logger
 
   def fetch_public_key(realm) do
-    with {:ok, client} <- Database.connect(realm: realm),
+    keyspace_name = CQLUtils.realm_name_to_keyspace_name(realm, Config.astarte_instance_id!())
+
+    with {:ok, client} <- Database.connect(realm: keyspace_name),
```

Comment:
You're going to hate me for this, _but_: I don't understand why you didn't just put the translation _inside_ the `Database.connect` implementation.

This would have solved 2 problems:
1. You wouldn't have had to touch the calling code
2. The current code has messed up naming: you're passing a `realm` option but you're actually passing a keyspace. Given that now those things are diverging, we should be extremely clear about where we're dealing with the `realm_name` and where we're dealing with the `keyspace`, otherwise we're causing unnecessary mental load when reading the code.

So my suggestion is just moving the translation inside `Database.connect` for all this sorts of call. Reverting shouldn't be too hard (the calling code basically remaing as-is and you just call `CQLUtils` in `astarte_data_access`).

Note that if the reason is "But `Config.astarte_instance_id()` comes from the application" then the second question is: why? That too should live in `astarte_data_access` and just `defdelegate`d in each `Config` application module (see, e.g., what we do for [`xandra_nodes`](https://github.com/astarte-platform/astarte/blob/4ebe50b4b17022ca60e4e83d9fc5f50526ed25b5/apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/config.ex#L397))